version: '3'
services:
  backend:
    build: ./backend
    ports:
      - "8000:8000"
    volumes:
      - ./backend/app:/app
      - ./backend/app/uploads:/app/uploads
      - ./backend/app/results:/app/results
    depends_on:
      - redis
      - postgres
    environment:
      - CELERY_BROKER_URL=redis://redis:6379/0
      - CELERY_RESULT_BACKEND=redis://redis:6379/0
      - DATABASE_URL=postgresql://automsa:automsa123@postgres/automsa_db
    networks:
      - app-network

  redis:
    image: redis:alpine
    ports:
      - "6379:6379"
    networks:
      - app-network

  celery_worker:
    build: ./backend
    command: celery -A celery_worker worker --loglevel=info --concurrency=1
    volumes:
      - ./backend/app:/app
    working_dir: /app
    depends_on:
      - redis
      - postgres
      - backend
    environment:
      - CELERY_BROKER_URL=redis://redis:6379/0
      - CELERY_RESULT_BACKEND=redis://redis:6379/0
      - PYTHONPATH=/app
      - DATABASE_URL=postgresql://automsa:automsa123@postgres/automsa_db
    networks:
      - app-network

  frontend:
    platform: linux/amd64
    build: ./frontend
    ports:
      - "3000:3000"
    volumes:
      - ./frontend:/app
      - /app/node_modules
    environment:
      - VITE_API_URL=http://localhost:8000
    depends_on:
      - backend
    networks:
      - app-network

  postgres:
    image: postgres:15
    environment:
      POSTGRES_USER: automsa
      POSTGRES_PASSWORD: automsa123
      POSTGRES_DB: automsa_db
    ports:
      - "5432:5432"
    volumes:
      - postgres_data:/var/lib/postgresql/data
    networks:
      - app-network

networks:
  app-network:
    driver: bridge 

volumes:
  postgres_data: 